### PR TITLE
[ApiBundle] Unification of locale in translations

### PR DESCRIPTION
**Pull request by Wojdylak**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | N?A
| License         | MIT

### CODE REVIEW

**Review by NoResponseMate — DISMISSED** [review]

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, Karol! :tada:

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductAssociationTypesContext.php [hunk 1]
```diff
@@ -167,7 +166,7 @@ public function iWantToModifyTheProductAssociationType(ProductAssociationTypeInt
      */
     public function iRenameItToIn(string $name, string $language): void
```
Comment (diimpp) [inline src/Sylius/Behat/Context/Api/Admin/ManagingProductAssociationTypesContext.php:167]:
```patch
--- public function iRenameItToIn(string $name, string $language): void
+++ public function iRenameItToIn(string $name, string $localeCode): void
```

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductAssociationTypesContext.php [hunk 2]
```diff
@@ -167,7 +166,7 @@ public function iWantToModifyTheProductAssociationType(ProductAssociationTypeInt
...
     {
-        $this->client->updateRequestData(['translations' => [$language => ['name' => $name, 'locale' => $language]]]);
+        $this->client->updateRequestData(['translations' => [$language => ['name' => $name]]]);
```
Comment (diimpp) [inline src/Sylius/Behat/Context/Api/Admin/ManagingProductAssociationTypesContext.php:169]:
```patch
--- $this->client->updateRequestData(['translations' => [$language => ['name' => $name]]]);
+++ $this->client->updateRequestData(['translations' => [$localeCode => ['name' => $name]]]);
```

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductOptionsContext.php [hunk 3]
```diff
@@ -61,7 +61,7 @@ public function iWantToModifyProductOption(ProductOptionInterface $productOption
      */
     public function iNameItInLanguage(?string $name = null, ?string $language = 'en_US'): void
     {
-        $data = ['translations' => [$language => ['locale' => $language]]];
+        $data = ['translations' => [$language => []]];
```
Comment (diimpp) [inline src/Sylius/Behat/Context/Api/Admin/ManagingProductOptionsContext.php:64]:
```patch
--- $data = ['translations' => [$language => []]];
+++ $data = ['translations' => [$localeCode => []]];
```

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductOptionsContext.php [hunk 4]
```diff
@@ -74,15 +74,15 @@ public function iNameItInLanguage(?string $name = null, ?string $language = 'en_
      */
     public function iRenameItInLanguage(string $name, string $language): void
     {
-        $this->client->updateRequestData(['translations' => [$language => ['name' => $name, 'locale' => $language]]]);
+        $this->client->updateRequestData(['translations' => [$language => ['name' => $name]]]);
```
Comment (diimpp) [inline src/Sylius/Behat/Context/Api/Admin/ManagingProductOptionsContext.php:77]:
```patch
--- $this->client->updateRequestData(['translations' => [$language => ['name' => $name]]]);
+++ $this->client->updateRequestData(['translations' => [$localeCode => ['name' => $name]]]);
```

File: src/Sylius/Behat/Context/Api/Admin/ManagingProductOptionsContext.php [hunk 5]
```diff
@@ -74,15 +74,15 @@ public function iNameItInLanguage(?string $name = null, ?string $language = 'en_
...
     }
 
     /**
      * @When I remove its name from :language translation
      */
     public function iRemoveItsNameFromTranslation(string $language): void
     {
-        $this->client->updateRequestData(['translations' => [$language => ['name' => '', 'locale' => $language]]]);
+        $this->client->updateRequestData(['translations' => [$language => ['name' => '']]]);
```
Comment (diimpp) [inline src/Sylius/Behat/Context/Api/Admin/ManagingProductOptionsContext.php:85]:
```patch
---  $this->client->updateRequestData(['translations' => [$language => ['name' => '']]]);
+++ $this->client->updateRequestData(['translations' => [$localeCode => ['name' => '']]]);
```

File: src/Sylius/Bundle/ApiBundle/spec/Serializer/TranslatableLocaleKeyDenormalizerSpec.php [hunk 6]
```diff
@@ -46,34 +46,37 @@ function it_does_nothing_if_there_is_no_translation_key(
...
     }
+
+    function it_throws_an_exception_if_locale_is_not_the_same_as_key(
+        DenormalizerInterface $denormalizer,
+    ): void {
+        $this->setDenormalizer($denormalizer);
+
+        $this
+            ->shouldThrow(TranslationLocaleMismatchException::class)
+            ->during('denormalize', [['translations' => ['de_DE' => ['locale' => 'locale']]], TranslatableInterface::class])
```
Comment (diimpp) [inline src/Sylius/Bundle/ApiBundle/spec/Serializer/TranslatableLocaleKeyDenormalizerSpec.php:79]: As idea for more readability. (I assume test is about two valid locales mismatch)
```patch
--- ->during('denormalize', [['translations' => ['de_DE' => ['locale' => 'locale']]], TranslatableInterface::class])
+++ ->during('denormalize', [['translations' => ['de_DE' => ['locale' => 'en_US']]], TranslatableInterface::class])
```

Comment (diimpp) [reply src/Sylius/Bundle/ApiBundle/spec/Serializer/TranslatableLocaleKeyDenormalizerSpec.php:79]: @Wojdylak cc

Comment (Wojdylak) [reply src/Sylius/Bundle/ApiBundle/spec/Serializer/TranslatableLocaleKeyDenormalizerSpec.php:79]: It looks better

**Pull request merged**